packages/hash/frontend/src/blocks/page/ComponentView.tsx

```diff
@@ -196,20 +204,124 @@ export class ComponentView implements NodeView<Schema> {
     }
   }
 
-  editableRef = (editableNode: HTMLElement) => {
-    if (
-      this.contentDOM &&
-      editableNode &&
-      !editableNode.contains(this.contentDOM)
-    ) {
-      editableNode.appendChild(this.contentDOM);
-      this.contentDOM.style.display = "";
+  private getDraftBlockEntity() {
+    const draftId = this.getBlockDraftId();
+    const entity = this.store.draft[draftId];
+
+    if (!entity || !isDraftBlockEntity(entity)) {
+      throw new Error("Component view can't find block entity");
+    }
+
+    return entity;
+  }
+
+  private getBlockDraftId() {
+    return this.editorView.state.doc.resolve(this.getPos()).node(2).attrs
+      .draftId;
+  }
+
+  private onBlockLoaded = () => {
+    /**
+     * After two calls to setImmediate, we know the block will have had the
+     * chance to initiate an editable section and thereby become selected if
+     * initiated via the suggester. If it hasn't happened in that time, we want
+     * to expire the opportunity to become automatically selected.
+     *
+     * @todo find a better way of knowing there's been the opportunity for
+     *       this which doesn't depend on timing
+     */
+    if (this.wasSuggested) {
```

Comment:
This may relate to my `wasSuggested` comment above, but I think we could usefully explain what 'automatically selected' is somewhere.

Reply:
Could you check if this is clearer (combined with my new comment on `wasSuggested`)?

[`e1db5b3` (#872)](https://github.com/hashintel/hash/pull/872/commits/e1db5b3c8515bb32f68c89a44f1004d6e6175912)

Reply:
It is, thank you